compute_sdk/tests/unit/test_serialization.py

```diff
@@ -619,3 +621,57 @@ def deserialize(self, payload):
 
     for e in (pyt_exc1, pyt_exc2):
         assert "is not a known serialization strategy" in str(e)
+
+    SerializationStrategy._CACHE.pop(NewStrategy.identifier)
+
+
+@pytest.mark.parametrize("wrong_length_id", ["", "1", "12", "1234"])
+def test_init_subclass_requires_correct_length(wrong_length_id):
+    with pytest.raises(ValueError) as pyt_exc:
+
+        class NewStrategy(SerializationStrategy):
+            identifier = wrong_length_id
+            for_code = True
+
+            def serialize(self, data):
+                pass
+
+            def deserialize(self, payload):
+                pass
+
+    assert "must be 3 characters long" in str(pyt_exc.value)
+
+
+def test_init_subclass_requires_newline():
+    with pytest.raises(ValueError) as pyt_exc:
+
+        class NewStrategy(SerializationStrategy):
+            identifier = "123"
+            for_code = True
+
+            def serialize(self, data):
+                pass
+
+            def deserialize(self, payload):
+                pass
+
+    assert "must end with a newline character" in str(pyt_exc.value)
+
+
+@pytest.mark.parametrize(
+    "existing_id", [s.identifier for s in SerializationStrategy._CACHE.values()]
+)
+def test_init_subclass_requires_unique_identifier(existing_id):
```

Comment:
Ensuring that all items in `_CACHE` raise this?  Heh.  One wasn't good enough for ya?  ;-)

Reply:
I get paid per test case 🚀